pairviewer.js

```diff
@@ -1235,91 +1149,70 @@ function handleUnusedPoints() {
   if (d.filtered === "false") {
     return 0;
   }
 
   let centerPos = proj.invert([fixedWidth / 2, fixedHeight / 2]),
-    start,
-    end,
-    distancePair; // distance of a flyer (in radians)
-
-  // function is called on 2 different data structures..
-  if (d.source) {
-    start = d.source;
-    end = d.target;
-  } else {
-    start = d.coordinates1;
-    end = d.coordinates2;
-  }
-
-  distancePair = d3.geoDistance(start, end);
+    start = d.source,
+    end = d.target,
+    distancePair = d3.geoDistance(start, end); // distance of a flyer (in radians)
 
-  let start_dist = 1.57 - d3.geoDistance(start, centerPos),
-    end_dist = 1.57 - d3.geoDistance(end, centerPos);
+  let startDist = 1.57 - d3.geoDistance(start, centerPos),
+    endDist = 1.57 - d3.geoDistance(end, centerPos);
 
   let fade = d3
     .scaleLinear()
     .domain([-0.1, 0])
     .range([0, 0.15]);
 
-  let dist = start_dist < end_dist ? start_dist : end_dist;
+  let dist = startDist < endDist ? startDist : endDist;
 
-  if(distancePair>=1.7) {
-    // 1.7 radians has been taken as an approximate value for a "long" flyer.
-    // However, this value can be changed as desired.
-    let avgLongitudeStartEnd = (start[0]+end[0])/2, // average longitude of the flyer's start and end points
-    avgLatitudeStartEnd = (start[1]+end[1])/2,      // average latitude of the flyer's start and end points
-    diffAvgCenterLongitude,                         // difference between avgLongitudeStartEnd and the longitude in centerPos
-    diffAvgCenterLatitude,                          // difference between avgLatitudeStartEnd and the latitude in centerPos
-    atBack = false;                                 // boolean representing if the flyer is at the back of the globe
-
-    diffAvgCenterLongitude = avgLongitudeStartEnd - centerPos[0];
-    if (diffAvgCenterLongitude > 180) {
-      diffAvgCenterLongitude = -180 + (diffAvgCenterLongitude-180);
-    }
-    else if (diffAvgCenterLongitude < -180) {
-      diffAvgCenterLongitude = 180 - (diffAvgCenterLongitude+180);
-    }
+  if (distancePair >= 1.7) {
+    /* 1.7 radians has been taken as an approximate value for a "long" flyer.
+     However, this value can be as desired.*/
+    let avgLongitudeStartEnd = (start[0] + end[0]) / 2,
+    avgLatitudeStartEnd = (start[1] + end[1]) / 2,
+    diffAvgCenterLongitude = avgLongitudeStartEnd - centerPos[0],
+    diffAvgCenterLatitude = avgLatitudeStartEnd - centerPos[1],
+    atBack = false; // boolean representing if the flyer is at the back of the globe
 
-    diffAvgCenterLatitude = avgLatitudeStartEnd - centerPos[1];
-    if (diffAvgCenterLatitude > 180) {
-      diffAvgCenterLatitude = -180 + (diffAvgCenterLatitude-180);
-    }
-    else if (diffAvgCenterLatitude < -180) {
-      diffAvgCenterLatitude = 180 - (diffAvgCenterLatitude+180);
-    }
+    diffAvgCenterLongitude = (diffAvgCenterLongitude + 180) % 360 - 180;
+    diffAvgCenterLatitude = (diffAvgCenterLatitude + 180) % 360 - 180;
```

Comment:
I think you should rather have + 540 instead of + 180 so that you can guarantee that the 2 values are in the range [-180, 180] (since a negative number modulo something in JavaScript is still negative)

Reply:
Ah yes I forgot... This discrepancy between Python and JS always confuses me.
(BTW technically, `%` in JavaScript is a _remainder_ operator, as it takes the sign of the dividend instead of the divisor. There is unfortunately no native _modulo_ operator in JS)